src/edit/DistortableCollection.Edit.js

```diff
@@ -0,0 +1,226 @@
+
+  _deselectAll: function(e) {
+    var oe;
+      
+    if (e) { oe = e.originalEvent; }
+    /** 
+     * prevents image deselection following the 'boxzoomend' event - note 'shift' must not be released until dragging is complete
+     * also prevents deselection following a click on a disabled img by differentiating it from the map
+     */
+    if (oe && (oe.shiftKey || oe.target instanceof HTMLImageElement)) { return; }
+  
+    this._group.eachLayer(function(layer) {
+      var edit = layer.editing;
+      L.DomUtil.removeClass(layer.getElement(), 'selected');
+      edit._deselect();
+    });
+
+    this._removeToolbar();
+
+    if (e) { L.DomEvent.stopPropagation(e); }
+  },
+
+   _unlockGroup: function() {
+    var map = this._group._map;
+
+    this._group.eachLayer(function (layer) {
+      if (this._group.isSelected(layer)) {
+        var edit = layer.editing;
+        if (edit._mode === 'lock') { 
+          map.removeLayer(edit._handles[edit._mode]); 
+          edit._unlock();
+          edit._refreshPopupIcons();
+        }
+      }
+    }, this);
+  },
+
+  _lockGroup: function() {
+    var map = this._group._map;
+
+    this._group.eachLayer(function (layer) {
+      if (this._group.isSelected(layer) ) {
+        var edit = layer.editing;
+        if (edit._mode !== 'lock') {
+          edit._lock();
+          map.addLayer(edit._handles[edit._mode]);
+          edit._refreshPopupIcons();
+          // map.addLayer also deselects the image, so we reselect here
+          L.DomUtil.addClass(layer.getElement(), 'selected');
+        }
+      }
+    }, this);
+  },
+
+  _addSelections: function(e) {
+    var box = e.boxZoomBounds,
+        map = this._group._map;
+
+    this._group.eachLayer(function(layer) {
+      var edit = layer.editing;
+      if (edit._selected) {
+        edit._deselect();
+      }
+
+      var imgBounds = L.latLngBounds(layer.getCorner(2), layer.getCorner(1));
+      imgBounds = map._latLngBoundsToNewLayerBounds(imgBounds, map.getZoom(), map.getCenter());
+      if (box.intersects(imgBounds) && edit.enabled()) {
+        if (!this.toolbar) {
+          this._addToolbar();
+        }
+        L.DomUtil.addClass(layer.getElement(), 'selected');
+      }
+    }, this);
+  },
+  
+  _removeGroup: function(e) {
+    var layersToRemove = this._group._toRemove(),
+        n = layersToRemove.length;
+
+    if (n === 0) { return; }
+    var choice = L.DomUtil.confirmDeletes(n);
+
+    if (choice) {
+      layersToRemove.forEach(function(layer) {
+        this._group.removeLayer(layer);
+      }, this);
+      this._removeToolbar();
+    }
+
+    if (e) { L.DomEvent.stopPropagation(e); }
+  },
+
+  _addToolbar: function() {
+    var map = this._group._map;
+
+    try {
+      if (!this.toolbar) {
+        this.toolbar = L.distortableImage.controlBar({
+          actions: this.editActions,
+          position: 'topleft'
+        }).addTo(map, this);
+        this.fire('toolbar:created');
+      }
+    } catch (e) { }
```

Comment:
I guess JS was throwing an error for `this.toolbar` not being initialized for this if that's why we have this in a try block? If that's the case then why not check "in" the object (`this`) keys for the `toolbar` property? That should avoid the try catch blocks and keep things simple.

Also, is suppressing the errors here okay? Maybe we could send these to the error logs? Just a thought, what do you think?

Reply:
@rexagod valid point but thats not really new code so if we can  open a new issue and discuss it there that would be great! 

Honestly not even sure what the issue was anymore it was a while ago. you can try rewriting and see if it works?

Reply:
Sure! I'll look into this.